### ISSUE TITLE
Add @HeadersMap

### ISSUE THREAD

**Issue by artshell** [square/retrofit#1736]
**> First:**
```java
    public interface ApiService {
        @Headers({
                "User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/49.0.2623.110 Safari/537.36",
                "Accept: text/html,text/plain,application/json",
                "Accept-Language: zh-cn,zh,en-US,en",
                "Accept: text/html,text/plain,application/json",
                "Accept-Language: zh-cn,zh,en-US,en",
                "Accept-Charset: utf-8,GB2312",
                "Connection: keep-alive"
        })
        @GET("Api-{module}-{method}")
        Call<TokenEntity> asyncGet(@Path("module") String pModule, @Path("method") String pMethod, @QueryMap Map<String, String> pParameterOptions);
    }
```
"Headers" modify to "HeaderMap" easy reading

**> Second:**
    This feature is exist or implement new feature?

```java
    private static class SingletonHolder {
        private static final OkHttpClient S_CLIENT;
        private static final Retrofit.Builder S_BUILDER;

        static {
            OkHttpClient.Builder nClientBuilder = new OkHttpClient.Builder();
            /* 连接超时2分钟 */
            nClientBuilder.connectTimeout(2 * 60 * 1000, TimeUnit.MILLISECONDS);
            /* 数据读取超时10分钟 */
            nClientBuilder.readTimeout(10 * 60 * 1000, TimeUnit.MILLISECONDS);
            /* 数据上传超时10分钟 */
            nClientBuilder.writeTimeout(10 * 60 * 1000, TimeUnit.MILLISECONDS);
            /* 连接失败后尝试重新连接 */
            nClientBuilder.retryOnConnectionFailure(true);
            S_CLIENT = nClientBuilder.build();

            /* Retrofit结合OkHttpClient*/
            S_BUILDER = new Retrofit.Builder();
            S_BUILDER.baseUrl("http://ybirds.lp/").addConverterFactory(GsonConverterFactory.create()).validateEagerly(true).client(S_CLIENT);
        }
    }

    public static Retrofit.Builder getRetrofitBuilder(){
        return SingletonHolder.S_BUILDER;
    }

    public interface ApiService {
        @GET("Api-{module}-{method}")
        <T> Call<T> asyncGet(@Path("module") String pModule, @Path("method") String pMethod);
    }

    Retrofit nRetrofit = getRetrofitBuilder().build();
    ApiService nApiService =  nRetrofit.create(ApiService.class);
    Map<String, String> nParameters = new HashMap<>();
    nParameters.put("appid","9CA630FF6AA2EE3E99");

    Call<TokenEntity> nTokenCall = nApiService.asyncGet("Token", "getToken", nParameters);
    // ------------------------------------
    Call<UserInfoEntity> nTokenCall = nApiService.asyncGet("user", "getUserInfo", nParameters);
    // and so on
```

**Comment by JakeWharton** [comment]
I'm not opposed to `@HeadersMap`.

**Comment by JakeWharton** [comment]
Merged as adfe99884bfc01dd3a1164a768fef96d6550208b.

**Comment by artshell** [comment]
Thanks, @JakeWharton
